src/fudge/log.clj

```diff
@@ -118,40 +130,62 @@
 
 ;; # Main logging functions
 
-(defn log
-  "Log a record with the config, level, and data provided"
-  [c level data]
-  (let [record (invoke c :prepare-fn data level)]
+(defn log!
```

Comment:
Why the exclamation mark?

Reply:
It shouldn't be called `log!` IMO as that goes against the Clojure standard. If I see ! at the end of a function, I expect it to be unsafe with respect to an STM transaction, not just impure.

Reply:
As in https://github.com/bbatsov/clojure-style-guide#changing-state-fns-with-exclamation-mark ?

That only says STM-unsafe should end with `!`, not that nothing else should.  But yeah, as above, I don't love the name anyway.

Reply:
I'm open to other suggestions. I basically nicked this pattern from `timbre` ( https://github.com/ptaoussanis/timbre ).